webdriver-spec.html

```diff
@@ -5340,26 +5402,146 @@ <h3 id="character-types">Character types</h3>
+ before the <a href=https://html.spec.whatwg.org/#event-loop>event loop</a>
+ is <a href=https://html.spec.whatwg.org/#pause>unpaused</a>
+ and control is returned to the <a>current top-level browsing context</a>.
+
+<p>By default <a href=https://html.spec.whatwg.org/#user-prompts>user prompts</a>
+ that are not handled automatically
+ unless a <a>user prompt handler</a> has been defined.
+ When a <a href=https://html.spec.whatwg.org/#user-prompts>user prompt</a> appears,
+ it is the task of the subsequent <a>command</a> to handle it.
+ If the subsequent requested <a>command</a> is not one listed in this chapter,
+ an <a>unexpected alert open</a> <a>error</a> will be returned.
+
+<p><a href=https://html.spec.whatwg.org/#user-prompts>User prompts</a>
+ that are spawned from <code>onbeforeunload</code> event handlers,
+ are <a>dismissed</a> implicitly upon <a href=#>navigation</a>,
+ regardless of the defined <a>user prompt handler</a>.
+
+<p>The following <dfn>table of simple dialogues</dfn> enumerates all
+ supported <a href=https://html.spec.whatwg.org/#simple-dialogs>simple dialogues</a>,
+ along with the <a>commands</a> that are allowed to interact with it
+ as a non-normative reference:
 
-<p class=issue><strong>Warning:</strong>
- This section has not yet been redefined to match the <a>routing requests</a> model,
- and uses old concepts and definitions.
- Please do not rely on it yet.
+<table class=simple>
+ <tr>
+  <th>Definition
+  <th>Dialogue
+  <th>Interactions
+ </tr>
 
-<p>This section describes how modal dialogs
- should be handled using the WebDriver API.
+ <tr>
+  <td><dfn>Alert</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-alert><code>alert</code></a>
+  <td>
+   <a>Accept Alert</a><br />
+   <a>Dismiss Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
 
-<section>
-<h3>window.alert, window.prompt and window.confirm</h3>
+ <tr>
+  <td><dfn>Confirm</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-confirm><code>confirm</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
+
+ <tr>
+  <td><dfn>Prompt</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-prompt><code>prompt</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a><br />
+   <a>Send Alert Text</a>
+ </tr>
+</table>
+
+<p class=issue>It’s not clear what interoperability looks like
+ for the ability to switch top-level browsing contexts
+ whilst a user prompt is present.
+
+<p>The <dfn>current user prompt</dfn> is said to be the active
+ <a href=https://html.spec.whatwg.org/#user-prompts>user prompt</a>,
+ which can be one of the entries on the <a>table of simple dialogues</a>,
+ the window.<a href=https://html.spec.whatwg.org/#dom-print><code>print</code></a> dialogue,
+ or a dialogue spawned by <a href=https://html.spec.whatwg.org/#dom-showmodaldialog><code>showModalDialog()</code></a>
```

Comment:
Gecko hasn’t yet though, and it will keep working there for ESR 38 until May 2016.  According to MDN, it is still also available in Chrome.